libpod/container_internal.go

```diff
@@ -1415,6 +1409,11 @@ func (c *Container) restartWithTimeout(ctx context.Context, timeout uint) (retEr
 		if err := c.stop(timeout); err != nil {
 			return err
 		}
+		if c.config.HealthCheckConfig != nil {
```

Comment:
Are we no longer doing this on stop? That doesn't seem safe?

Reply:
We're doing it on cleanup and restart.  I added it to stop to fix a symptom that should have been added to restart. Having it in stop will kill the process stopping.